modules/web-console/web-console-server/src/main/java/org/apache/ignite/console/web/socket/WebSocketManager.java

```diff
@@ -185,11 +200,40 @@ private void ping(WebSocketSession ws) {
      * @throws IOException If failed to process.
      */
     private void registerAgent(WebSocketEvent evt, WebSocketSession ws) throws IOException {
-        AgentInfo agentInfo = fromJson(evt.getPayload(), AgentInfo.class);
+        AgentHandshakeRequest req = fromJson(evt.getPayload(), AgentHandshakeRequest.class);
+
+        AgentHandshakeResponse res = new AgentHandshakeResponse();
+
+        if (F.isEmpty(req.getTokens()))
+            res.setError("Tokens not set. Please reload agent or check settings.");
+        else if (!F.isEmpty(req.getVersion()) && !F.isEmpty(req.getBuildTime()) & !F.isEmpty(supportedAgents)) {
+            // TODO WC-1053 Implement version check in beta2 stage.
+            res.setError("You are using an older version of the agent. Please reload agent.");
+        }
+        else {
+            Set<String> tokens = req.getTokens();
+
+            Set<String> validTokens = accRepo.validateTokens(tokens);
 
-        log.info("Agent connected: " + agentInfo);
+            if (F.isEmpty(validTokens)) {
+                res.setError("Failed to authenticate with token(s): " + tokens + "." +
+                    " Please reload agent or check settings.");
+            }
+            else {
+                req.setTokens(validTokens);
+                res.setTokens(validTokens);
+            }
+        }
+
+        if (F.isEmpty(res.getError())) {
+            log.info("Agent connected: " + req);
+
+            agents.put(ws, req);
```

Comment:
Why we save full request object if we use only tokens?

Reply:
Fixed.